explainaboard/utils/eval_utils/qa_table_text_hybrid.py

```diff
@@ -0,0 +1,328 @@
+    text = white_space_fix(text)
+    tokens = text.split()
+    tokens = [token.strip(STRIPPED_CHARACTERS) for token in tokens]
+    return tokens
+
+
+def _answer_to_bags(answer: Union[str, list[str], tuple[str, ...]]):
+    if isinstance(answer, (list, tuple)):
+        raw_spans = answer
+    else:
+        raw_spans = [answer]
+    normalized_spans: list[str] = []
+    token_bags = []
+    for raw_span in raw_spans:
+        normalized_span = normalize_answer(raw_span)
+        normalized_spans.append(normalized_span)
+        token_bags.append(set(normalized_span.split()))
+    return normalized_spans, token_bags
+
+
+def _align_bags(predicted: list[set[str]], gold: list[set[str]]) -> list[float]:
+    """
+    Takes gold and predicted answer sets and first finds the optimal 1-1 alignment
+    between them and gets maximum metric values over all the answers.
+    """
+    scores = np.zeros([len(gold), len(predicted)])
+    for gold_index, gold_item in enumerate(gold):
+        for pred_index, pred_item in enumerate(predicted):
+            # if _match_numbers_if_present(gold_item, pred_item): no
+            # need to match number in tatqa
+            scores[gold_index, pred_index] = _compute_f1(pred_item, gold_item)
+    row_ind, col_ind = linear_sum_assignment(-scores)
+
+    max_scores = np.zeros([max(len(gold), len(predicted))])
+    for row, column in zip(row_ind, col_ind):
+        max_scores[row] = max(max_scores[row], scores[row, column])
+    return max_scores
+
+
+def _compute_f1(predicted_bag: set[str], gold_bag: set[str]) -> float:
+    intersection = len(gold_bag.intersection(predicted_bag))
+    if not predicted_bag:
+        precision = 1.0
+    else:
+        precision = intersection / float(len(predicted_bag))
+    if not gold_bag:
+        recall = 1.0
+    else:
+        recall = intersection / float(len(gold_bag))
+    f1 = (
+        (2 * precision * recall) / (precision + recall)
+        if not (precision == 0.0 and recall == 0.0)
+        else 0.0
+    )
+    return f1
+
+
+def _match_numbers_if_present(gold_bag: set[str], predicted_bag: set[str]) -> bool:
+    gold_numbers = set()
+    predicted_numbers = set()
+    for word in gold_bag:
+        if is_number(word):
+            gold_numbers.add(word)
+    for word in predicted_bag:
+        if is_number(word):
+            predicted_numbers.add(word)
+    if (not gold_numbers) or gold_numbers.intersection(predicted_numbers):
+        return True
+    return False
+
+
+def extract_gold_answers(qa_annotation):
+    '''
+    span
+    multi-span
+    arithmetic (+ - * /)
+    count
+    date
+    other
+    gold answers is a list of list, each item in gold answers is a valid answer
+    '''
+    answer_type, scale = qa_annotation["answer_type"], qa_annotation['scale']
+    answer_content = qa_annotation['answer']
+    gold_answers = []
+    if answer_type in ['multi-span', 'span']:  # list
+        assert isinstance(answer_content, list), answer_content
```

Comment:
Assertions can't be used in library code. It doesn't work with several runtime options.
```suggestion
        if not isinstance(answer_content, list):
            raise ValueError("answer_content must be list.")
```